soc/riscv/riscv-privilege/gd32vf103/Kconfig.defconfig.gd32vf103

```diff
@@ -10,10 +10,7 @@ config KERNEL_ENTRY
 	default "__nuclei_start"
 
 config SYS_CLOCK_HW_CYCLES_PER_SEC
-	default $(dt_node_int_prop_int,/cpus/cpu@0,clock-frequency) if RISCV_MACHINE_TIMER
```

Comment:
`SYS_CLOCK_HW_CYCLES_PER_SEC` is system clock value, on gd32vf103, it is 108MHz. 

From below code, `SYS_CLOCK_HW_CYCLES_PER_SEC` can effect the system timer behavior, we can check why `CYC_PER_TICK` can not work properly on gd32vf103.
 
```
// drivers/timer/riscv_machine_timer.c:76

#define CYC_PER_TICK (uint32_t)(sys_clock_hw_cycles_per_sec() \
				/ CONFIG_SYS_CLOCK_TICKS_PER_SEC)
```

Reply:
> SYS_CLOCK_HW_CYCLES_PER_SEC is system clock value, on gd32vf103, it is 108MHz.

```
config SYS_CLOCK_HW_CYCLES_PER_SEC
        int "System clock's h/w timer frequency"
        help
          This option specifies the frequency of the hardware timer used for the
          system clock (in Hz). This option is set by the SOC's or board's Kconfig file
          and the user should generally avoid modifying it via the menu configuration.
```

The description is define it as `frequency of the hardware timer`.

As an actual example, even if the CPU frequency is 150MHz like SiFive FE310, SYS_CLOCK_HW_CYCLES_PER_SEC is set to the clock frequency of 32768.

https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/riscv/riscv-privilege/sifive-freedom/Kconfig.defconfig.series#L8-L9

Reply:
`clock_control_gd32` driver also use `SYS_CLOCK_HW_CYCLES_PER_SEC`.

Reply:
> From below code, SYS_CLOCK_HW_CYCLES_PER_SEC can effect the system timer behavior, we can check why CYC_PER_TICK can not work properly on gd32vf103.

Before https://github.com/zephyrproject-rtos/zephyr/commit/11a2107d991b14a7abb7081498c914a9f57ed089 commit,
CYC_PER_TICK define as below,
```

#define CYC_PER_TICK ((uint32_t)((uint64_t) (sys_clock_hw_cycles_per_sec()			 \
					     >> CONFIG_RISCV_MACHINE_TIMER_SYSTEM_CLOCK_DIVIDER) \
				 / (uint64_t)CONFIG_SYS_CLOCK_TICKS_PER_SEC))
```

From this point of view, I think that there is no problem because it is only calculated in advance.

Reply:
I'm not sure what is the correct value here but @soburi is right here: `SYS_CLOCK_HW_CYCLES_PER_SEC` is different from the CPU clock frequency.

Reply:
From [GD32VF103_User_Manual](https://gd32mcu.com/en/download/6?kw=GD32VF1) `5.2. Clock control unit (CCTL)` as below.  Does the 27MHz value come from here? and implicit assume CK_SYS = CK_AHB = 108MHz?
```
The RISCV System Timer (SysTick) external clock is clocked with the AHB clock (HCLK)
divided by 4.
```

Reply:
Yes. More explicitly described in clock tree. (CK_CST)
https://www.gigadevice.com.cn/Public/Uploads/uploadfile/files/20230209/GD32VF103_User_Manual_EN_Rev1.4.pdf#%5B%7B%22num%22%3A166%2C%22gen%22%3A0%7D%2C%7B%22name%22%3A%22XYZ%22%7D%2C60%2C769%2C0%5D

Reply:
> clock_control_gd32 driver also use SYS_CLOCK_HW_CYCLES_PER_SEC.

Fixed clock_control_gd32 to directly refer to the cpu frequency defined in dts.